RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -0,0 +1,285 @@
+    mvaFileB_ = iConfig.getUntrackedParameter<edm::FileInPath>("mvaFileB");
+    mvaFileE_ = iConfig.getUntrackedParameter<edm::FileInPath>("mvaFileE");
+
+    mvaScaleMeanB_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleMeanB");
+    mvaScaleStdB_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleStdB");
+    mvaScaleMeanE_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleMeanE");
+    mvaScaleStdE_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleStdE");
+
+    mvaEstimator_ = std::make_pair(std::make_unique<SeedMvaEstimator>(mvaFileB_, mvaScaleMeanB_, mvaScaleStdB_),
+                                   std::make_unique<SeedMvaEstimator>(mvaFileE_, mvaScaleMeanE_, mvaScaleStdE_));
+  }
+
+  produces<TrajectorySeedCollection>();
+}
+
+// -- method called on each new Event
+void MuonHLTSeedMVAClassifier::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  auto result = std::make_unique<TrajectorySeedCollection>();
+
+  if (rejectAll_) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  if (doSort_ && nSeedsMaxB_ <= 0 && nSeedsMaxE_ <= 0) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  if (!doSort_ && mvaCutB_ > 1. && mvaCutE_ > 1.) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  edm::ESHandle<TrackerGeometry> trkGeom;
+  iSetup.get<TrackerDigiGeometryRecord>().get(trkGeom);
+
+  edm::Handle<l1t::MuonBxCollection> h_L1Muon;
+  iEvent.getByToken(t_L1Muon_, h_L1Muon);
+
+  edm::Handle<reco::RecoChargedCandidateCollection> h_L2Muon;
+  if (!isFromL1_) {
+    iEvent.getByToken(t_L2Muon_, h_L2Muon);
+  }
+
+  edm::Handle<TrajectorySeedCollection> h_Seed;
+  iEvent.getByToken(t_Seed_, h_Seed);
+
+  std::vector<std::pair<unsigned, double>> pairSeedIdxMvaScoreB = {};
+  std::vector<std::pair<unsigned, double>> pairSeedIdxMvaScoreE = {};
+  for (auto i = 0U; i < h_Seed->size(); ++i) {
+    const auto& seed(h_Seed->at(i));
+
+    GlobalVector global_p =
+        trkGeom->idToDet(seed.startingState().detId())->surface().toGlobal(seed.startingState().parameters().momentum());
+
+    bool isB = (std::abs(global_p.eta()) < etaEdge_);
+
+    if (doSort_) {
+      if (isB) {
+        if (nSeedsMaxB_ <= 0) {
+          continue;
+        }
+      } else {
+        if (nSeedsMaxE_ <= 0) {
+          continue;
+        }
+      }
+    } else {
+      if (isB) {
+        if (mvaCutB_ > 1.0) {
+          continue;
+        } else if (mvaCutB_ <= 0.) {
+          result->emplace_back(seed);
+          continue;
+        }
+      } else {
+        if (mvaCutE_ > 1.0) {
+          continue;
+        } else if (mvaCutE_ <= 0.) {
+          result->emplace_back(seed);
+          continue;
+        }
+      }
+    }
+
+    double mva = getSeedMva(mvaEstimator_, seed, global_p, h_L1Muon, minL1Qual_, h_L2Muon, isFromL1_, baseScore_);
+
+    double score = 1. / (1. + std::exp(-1. * mva));
+    bool passMva = ((isB && (score > mvaCutB_)) || (!isB && (score > mvaCutE_)));
```

Comment:
This is probably more compact and intuitive:
```suggestion
    bool passMva = isB? score > mvaCutB_ : score > mvaCutE_;
```